### PR TITLE
DBM version 1.4

### PR DESCRIPTION
**Pull request by xavierleroy**
- Autoconfiguration revised to find GDBM in more places (https://github.com/ocaml/dbm/pull/16)
- Add missing free in Dbm.fetch (https://github.com/ocaml/dbm/pull/18)

### CODE REVIEW

File: packages/dbm/dbm.1.4/opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ["./configure"]
+  [make "all"]
+  [make "test"] {with-test}
+]
+depends: [
+  "ocaml"
+  "ocamlfind" {build}
+  "conf-dbm"
+]
+install: [
```
Comment (mseri) [inline packages/dbm/dbm.1.4/opam:18]:
```suggestion
conflicts: [ "ocaml-option-bytecode-only" ]
install: [
```

Comment (mseri) [reply packages/dbm/dbm.1.4/opam:18]: For
```

#=== ERROR while compiling dbm.1.4 ============================================#
# context              2.3.0~alpha~dev | linux/x86_32 | ocaml-base-compiler.5.2.0 | pinned(https://github.com/ocaml/dbm/archive/camldbm-1.4.tar.gz)
# path                 ~/.opam/5.2/.opam-switch/build/dbm.1.4
# command              ~/.opam/opam-init/hooks/sandbox.sh build make all
# exit-code            2
# env-file             ~/.opam/log/dbm-7-5ff3ac.env
# output-file          ~/.opam/log/dbm-7-5ff3ac.out
### output ###
# ocamlc -c -ccopt "" -ccopt "" -ccopt -DCAML_NAME_SPACE cldbm.c
# ocamlmklib -oc camldbm cldbm.o -lgdbm
# ocamlc -c  dbm.mli
# ocamlc -c  dbm.ml
# ocamlmklib -o dbm -oc camldbm -linkall dbm.cmo -lgdbm
# ocamlopt -c  dbm.ml
# make: ocamlopt: No such file or directory
# make: *** [Makefile:51: dbm.cmx] Error 127
```

File: packages/dbm/dbm.1.4/opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  ["mkdir" "-p" "%{lib}%/dbm"]
+  [make "install" "STUBLIBDIR=%{lib}%/stublibs" "LIBDIR=%{lib}%/dbm"]
+  ["cp" "META" "%{lib}%/dbm"]
+]
+remove: [
+  ["rm" "-f" "%{lib}%/stublibs/dllcamldbm.so"]
+  ["ocamlfind" "remove" "dbm"]
+]
```
Comment (mseri) [inline packages/dbm/dbm.1.4/opam:26]:
```suggestion
```
This is no longer needed with opam 2

File: packages/dbm/dbm.1.4/opam [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+synopsis: "Binding to the NDBM/GDBM Unix \"databases\""
+flags: light-uninstall
```
Comment (mseri) [inline packages/dbm/dbm.1.4/opam:28]:
```suggestion
```
This is also no longer needed with opam 2

**Comment by xavierleroy** [comment]
Thanks for all the suggestions, much appreciated.  This library is legacy code... with a legacy OPAM file :-)

**Comment by mseri** [comment]
Thanks

**Pull request merged**